src/main/scala/cromwell/engine/TaskExecutionContext.scala

```diff
@@ -0,0 +1,3 @@
+package cromwell.engine
+
+trait TaskExecutionContext
```

Comment:
can you comment this heavily as to why it is necessary, how it is intended to work, etc? The level should be aimed at an implementer of Backend coming in this w/o much context